src/Microsoft.DotNet.Core.Build.Tasks/DependencyContextBuilder.cs

```diff
@@ -0,0 +1,270 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using Microsoft.Extensions.DependencyModel;
+using NuGet.Frameworks;
+using NuGet.Packaging.Core;
+using NuGet.ProjectModel;
+
+namespace Microsoft.DotNet.Core.Build.Tasks
+{
+    public class DependencyContextBuilder
+    {
+        public DependencyContextBuilder()
+        {
+        }
+
+        public DependencyContext Build(
```

Comment:
>  I understand that you are bringing this over from ProjectModel, is that correct?

Yes, this file started as a copy from the one in ProjectModel.

>  Is this the right time to fix some of this?

I don't understand this question.  We need to generate a .deps.json file without depending on the ProjectModel library, since that library is going to be deleted.

> Also, do we have any existing unit tests for this that we should bring along as well?

There is one class, but it can't be brought along because it is dependent on ProjectModel.